### PR TITLE
[KARAF-2511] Add karaf-dump-example

### PR DESCRIPTION
**Pull request by jbonofre**

### CODE REVIEW

File: examples/karaf-dump-example/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+```
+mvn clean install
+```
+
+## Deployment
+
+On a running Karaf instance, install the dump bundle:
+
+```
+karaf@root()> bundle:install -s
```
Comment (fpapon) [inline examples/karaf-dump-example/README.md]: Install bundle command should be `bundle:install -s mvn:org.apache.karaf.examples/karaf-dump-example/4.2.0-SNAPSHOT`?

File: examples/karaf-dump-example/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+You will find the dump zip file in the Karaf home folder. If you extract this zip, in addition of the regular files (heap dump, thread dump, ...),
+you will see screenshot files created by our example dump provider:
+
+```
+Archive:  xxxx-xx-xx_xxxxxx.zip
+  inflating: environment.txt         
+  inflating: memory.txt              
+  inflating: threads.txt             
+  inflating: heapdump.txt
```
Comment (fpapon) [inline examples/karaf-dump-example/README.md]: heapdump.txt is missing in the archive

File: examples/karaf-dump-example/README.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+## Usage
+
+Once you have installed the bundle, you can take a dump using:
+
+```
+karaf@root()> dev:dump-create
+```
+
+You will find the dump zip file in the Karaf home folder. If you extract this zip, in addition of the regular files (heap dump, thread dump, ...),
+you will see screenshot files created by our example dump provider:
```
Comment (fpapon) [inline examples/karaf-dump-example/README.md]: the screenshot files are missing in the archives

File: examples/karaf-dump-example/README.md (hunk 1 above)
Comment (fpapon) [reply examples/karaf-dump-example/README.md]: When starting the bundle there is an unresolved error 
```
karaf@root()> bundle:install -s mvn:org.apache.karaf.examples/karaf-dump-example/4.2.0-SNAPSHOT 
Bundle ID: 43
Error executing command: Error installing bundles:
	Unable to start bundle mvn:org.apache.karaf.examples/karaf-dump-example/4.2.0-SNAPSHOT: org.osgi.framework.BundleException: Unable to resolve org.apache.karaf.examples.karaf-dump-example [43](R 43.0): missing requirement [org.apache.karaf.examples.karaf-dump-example [43](R 43.0)] osgi.extender; (&(osgi.extender=osgi.component)(version>=1.3.0)(!(version>=2.0.0))) Unresolved requirements: [[org.apache.karaf.examples.karaf-dump-example [43](R 43.0)] osgi.extender; (&(osgi.extender=osgi.component)(version>=1.3.0)(!(version>=2.0.0)))]
```

File: examples/karaf-dump-example/README.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+```
+mvn clean install
+```
+
+## Deployment
+
+On a running Karaf instance, install the dump bundle:
+
+```
```
Comment (fpapon) [inline examples/karaf-dump-example/README.md]: We have to install scr feature before install the bundle, we should add this in the documentation
```
feature:install scr
````

**Comment by jbonofre** [comment]
Fixed.

**Comment by fpapon** [comment]
Good for me, can you squash please ?

**Pull request merged**